src/auto_archiver/base_modules/storage.py

```diff
@@ -4,40 +4,17 @@
 from typing import IO, Optional
 import os
 
-from ..utils.misc import random_str
+from auto_archiver.utils.misc import random_str
 
-from ..core import Media, Step, ArchivingContext, Metadata
-from ..enrichers import HashEnricher
+from auto_archiver.core import Media, Step, ArchivingContext, Metadata
+from auto_archiver.modules.hash_enricher.hash_enricher import HashEnricher
 from loguru import logger
 from slugify import slugify
 
 
 @dataclass
 class Storage(Step):
     name = "storage"
-    PATH_GENERATOR_OPTIONS = ["flat", "url", "random"]
-    FILENAME_GENERATOR_CHOICES = ["random", "static"]
-
-    def __init__(self, config: dict) -> None:
-        # without this STEP.__init__ is not called
-        super().__init__(config)
-        assert self.path_generator in Storage.PATH_GENERATOR_OPTIONS, f"path_generator must be one of {Storage.PATH_GENERATOR_OPTIONS}"
-        assert self.filename_generator in Storage.FILENAME_GENERATOR_CHOICES, f"filename_generator must be one of {Storage.FILENAME_GENERATOR_CHOICES}"
```

Comment:
This is awesome. We can remove so much boilerplate code now!  Just need `choices = ` in the manifest configs, and it'll do all the checks for us!